invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/Location/LocationForms/InternalLocationEditor/components/InternalLocationForm/InternalLocationForm.js

```diff
@@ -1,100 +1,64 @@
-        : await internalLocationApi.create(values);
-
-      setTimeout(() => {
-        this.props.sendSuccessNotification(
-          'Success!',
-          this.successSubmitMessage
-        );
-        goTo(BackOfficeRoutes.locationsList);
-      }, ES_DELAY);
-    } catch (error) {
-      const errors = getIn(error, 'response.data.errors', []);
+  updateInternalLocation = (pid, data) => {
+    return internalLocationApi.update(pid, data);
+  };
 
-      if (isEmpty(errors)) {
-        throw error;
-      } else {
-        const errorData = error.response.data;
-        const payload = {};
-        for (const fieldError of errorData.errors) {
-          payload[fieldError.field] = fieldError.message;
-        }
-        actions.setErrors(payload);
-        actions.setSubmitting(false);
-      }
-    }
+  createInternalLocation = async data => {
+    return internalLocationApi.create(data);
   };
 
+  successCallback = () => goTo(BackOfficeRoutes.locationsList);
+
   render() {
     return (
-      <>
-        <Grid>
-          <Grid.Row centered>
-            <Header>{this.title}</Header>
-          </Grid.Row>
-        </Grid>
-
-        <Formik
-          initialValues={
-            this.formInitialData
-              ? this.prepareData(this.formInitialData.metadata)
-              : {}
-          }
-          onSubmit={this.onSubmit}
-          render={({ isSubmitting, handleSubmit }) => (
-            <Form onSubmit={handleSubmit} loading={isSubmitting}>
-              <Segment>
-                <StringField label="Name" fieldPath="name" required />
-                {/* TODO make this an autocompletion field from `location.list` endpoint */}
-                <StringField
-                  label="Location pid"
-                  fieldPath="location_pid"
-                  required
-                />
-                <StringField
-                  label="Physical Location"
-                  fieldPath="physical_location"
-                />
-                <TextField
-                  label="Notes"
-                  fieldPath="notes"
-                  uiProps={{ rows: 5 }}
-                />
-              </Segment>
-              <Button color="green" disabled={isSubmitting}>
-                Submit
-              </Button>
-            </Form>
-          )}
-        />
-      </>
+      <BaseForm
+        initialValues={
+          this.formInitialData
+            ? this.prepareData(this.formInitialData.metadata)
+            : {}
+        }
+        editApiMethod={this.updateInternalLocation}
+        createApiMethod={this.createInternalLocation}
+        successCallback={this.successCallback}
+        successSubmitMessage={this.successSubmitMessage}
+        title={this.title}
+        pid={this.pid ? this.pid : undefined}
+      >
+        <StringField label="Name" fieldPath="name" required />
+        {/* TODO make this an autocompletion field from `location.list` endpoint */}
```

Comment:
Are you sure we will remember this TODO?

Reply:
I will create an issue for that!

Reply:
https://github.com/inveniosoftware/invenio-app-ils/issues/532